bundler/helpers/v2/monkey_patches/definition_ruby_version_patch.rb

```diff
@@ -10,7 +10,7 @@ def source_requirements
         Gem::Specification.new("Ruby\0", requested_version)
     end
 
-    %w(2.5.3 2.6.10 2.7.6 3.0.4).each do |version|
+    %w(2.5.3 2.6.10 2.7.6 3.0.4, 3.1.2).each do |version|
```

Comment:
I copied the existing format, but I'm unclear:
1. Why does the `bundler` `v1` version list specify the patch level? But `bundler` `v2` version list doesn't?
2. Beyond adding the most recent version, should I be updating the patch versions for the previous versions as well? For example `3.0.4` --> [`3.0.5`](https://www.ruby-lang.org/en/news/2022/11/24/ruby-3-0-5-released/)?

@deivid-rodriguez maybe you know?

Reply:
Regarding 1, I think that's "historical reasons". Up until Ruby 1.9.3, so like... 12 years ago, the Ruby version used to include a patch level and that was a meaningful part of the version, so Ruby would release versions like 1.9.3-p286, or 1.9.3-p327. Now we use only `x.y.z` versions without the `-p<n>` part.

The `-p<n>` part is still provided under the hood but is no longer meaningful. My guess is that Bundler 1.17.3 still needed that part of the version to be included even if not really necessary? Not sure, but I guess not worth touching that since we should be dropping Bundler 1 support soon-ish anyways.

Regarding 2, I think that should be ok. In the Ruby world, I've never seen any gems pinning up to the patch level of Ruby, normally they will specify things like `>= 3.0.0` or `~> 3.0`, so either should work for most cases.

Reply:
By the way, I don't think you need to add this here. At least not as part of this PR. The idea here is to add the latest patch levels of previous supported versions of Ruby to the current Ruby version, which is already considered.

Reply:
Suggestion per the above

```suggestion
    %w(2.5.3 2.6.10 2.7.6 3.0.4).each do |version|
```

Reply:
Again, many thanks @deivid-rodriguez. For simplicity I bumped to `2.7.7` and `3.0.5`, as it doesn't hurt if they're on latest... also omitted the `3.1.2` as you suggested.